modules/backend/clinic/staff/Staff.hpp

```diff
@@ -12,12 +13,19 @@ namespace staff
 class Staff
 {
   public:
-    void addDoctor(const doctor::Doctor& doctor);
-    std::vector<doctor::Doctor> getDoctors() const;
+    void addDoctor(doctor::DoctorPtr doctor);
+    doctor::iDoctor* getDoctor(const common::Name& name);
+    std::vector<doctor::iDoctor*> getDoctors();
+
+    void addReceptionist(receptionist::ReceptionistPtr receptionist);
+    receptionist::iReceptionist* getReceptionist(const common::Name& name);
+    std::vector<receptionist::iReceptionist*> getReceptionists();
 
   private:
-    std::vector<doctor::Doctor> doctors_;
-    // TODO introduce missing staff members
+    template <typename Person> typename Person::iterator getPersonByName(Person& person, const common::Name& name);
```

Comment:
If we want this function to be not visible as public, than it might be reasonable to hide completely it's existance.
Especially when it is templated.
Therefore I suggest to put it in cpp file in anonymous namespace and remove it from header.
I saw, that in cpp it is on top anyway so it's best place to put in into unnamed namespace.

Of course, if we predict that it might be public some day, than it is reasonable to keep it in header as well.

Reply:
From a readiness reason I keep declarations in hpp file, so once you check the header you know what's in the class

Reply:
+ it might turn out that this function could be moved to utilities, as I find it helpful in scope of Patient

Reply:
Sure, this approach is not bad after all.
I just wanted to consider all options.